release/models/interfaces/openconfig-if-sdn-ext.yang

```diff
@@ -47,11 +47,26 @@ module openconfig-if-sdn-ext {
         This is used by an external programming entity to disable an interface
         (usually part of an aggregate) for the purposes of forwarding
         traffic. This allows a logical aggregate to continue to be
-        used with partial capacity, for example.  Note that setting
-        `forwarding-viable = false` is not equivalent to
-        administratively disabling the interface -- in particular, the
-        interface is expected to participate in L2 protocols such as
-        LLDP or LACP even if it blocked from forwarding traffic.";
+        used with partial capacity. Setting `forwarding-viable = false` is not 
+        equivalent to administratively disabling the interface.
+        Some rules to follow when a port/bundle is set for 
+        Forwarding-viable=False:
+          1. Given that L2 protocols like LACP should be allowed to be 
+          transmitted and received on such ports, a device mustn't consider 
+          such ports in the calculation for minimum number of links required 
+          for an aggregate bundle when such a configuration exists on the box.
+          Hence, the device mustn't disable the bundle because the minimum
+          number of member links requirement is not met for the bundle.
+          2. L2 protocols like LLDP, LACP must be allowed for transmit and 
+          receive on such ports/bundles, IS-IS PDUs should be handled as per the 
+          requirements for L3 packets below.
+          3. L3 packets, control-plane as well as data-plane must be received
+          but not transmitted on such ports/bundles. By doing so, expectation
+          is that the other end of the connection ceases to use its respective
+          ports for transmit and receive when either the dead-timer or the
+          hold-down timer of its L3 protocol expires. 
```

Comment:
We should define what the expectation is for received L3 packets.  I would guess the device is supposed to forward L3 packets, right?    If yes, then stating something like the following might make this more clear?

4. Received L3 packets should be evaluated by the data plane as usual.  For example, when a dataplane packet is received, a lookup should be performed and the packet forwarded as usual.  If an L3 control plane packet is received, it should be processed as usual.

Reply:
updated as per the internal doc, ptal.

Reply:
Hi Sach, this statement appears tailored for a protocol like IS-IS. The peer does not _have to_ stop sending traffic to this device on the non-viable ports based on a timeout even if all the trunk members are not-viable. So, can we modify the statement to talk only about DUT and not enforce expectations on the peer? Specifically,
1. Stop L3 transmit for both control and data-plane packets.
2. L3 forwarding should work as usual on Rx without any timeout considerations. The peer can choose to continue sending this packets here indefinitely.

If you prefer, you can call out IS-IS as an example, where the timeout will happen.

Reply:
@nachikethas I have updated rule#3 to include other use cases, ptal